arches/app/models/graph.py

```diff
@@ -563,19 +564,18 @@ def delete(self):
                 )
             )
 
-    def delete_instances(self, verbose=False):
+    def delete_instances(self, userid=None, verbose=False):
         """
         deletes all associated resource instances
 
         """
-        if verbose is True:
-            bar = pyprind.ProgBar(Resource.objects.filter(graph_id=self.graphid).count())
-        for resource in Resource.objects.filter(graph_id=self.graphid):
-            resource.delete()
-            if verbose is True:
-                bar.update()
-        if verbose is True:
-            print(bar)
+
+        bulk_deleter = BulkDataDeletion()
+        loadid = uuid.uuid4()
+        resp = bulk_deleter.delete_resources(userid, loadid, self.graphid, verbose=verbose)
```

Comment:
When invoked from the command line using something like `python manage.py resources remove_resources -g graphid-here`, the resourceids argument is not provided and prevents users from deleting resources for a given graph. This is not an issue if you do not provide a graphid and delete all resources wholesale.

Reply:
This also appears to be an issue with the `index_resource_deletion` method below

Reply:
> When invoked from the command line using something like `python manage.py resources remove_resources -g graphid-here`, the resourceids argument is not provided and prevents users from deleting resources for a given graph. This is not an issue if you do not provide a graphid and delete all resources wholesale.

@johnatawnclementawn I'm not sure I understand. In the case you mention `remove_resources(graphid)` -> `graph.delete_instances()` -> `delete_resources(self.graphid)` and deletes all resources of that graph. Or perhaps I'm misreading your comment?

Reply:
Sorry for the confusion @whatisgalen - I'm just seeing an issue of backwards compatibility with the `remove_resources` [CLI command](https://arches.readthedocs.io/en/stable/developing/reference/command-line-reference/#other-data-management-commands), specifically when providing a graphid. Here's the traceback I see when I run that command with a specified graphid:

`All resources associated with the 'Concept' Resource Model will be removed. continue? [Y/n] y
Traceback (most recent call last):
  File "/web_root/arches_rdm_example_project/manage.py", line 30, in <module>
    execute_from_command_line(sys.argv)
  File "/web_root/ENV/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/web_root/ENV/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/web_root/ENV/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/web_root/ENV/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/web_root/arches/arches/management/commands/resources.py", line 49, in handle
    self.remove_resources(force=options["yes"], graphid=options["graph"])
  File "/web_root/arches/arches/management/commands/resources.py", line 71, in remove_resources
    graph.delete_instances(verbose=True)
  File "/web_root/arches/arches/app/models/graph.py", line 575, in delete_instances
    resp = bulk_deleter.delete_resources(userid, loadid, self.graphid, verbose=verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: BulkDataDeletion.delete_resources() missing 1 required positional argument: 'resourceids'`

Reply:
@johnatawnclementawn  That's perfect, thanks for providing the clarification and stack trace. I've included a dummy value of `None` for that pos arg `resourceids` so let me know if you're still seeing an issue testing this. Thanks!